builder/builder-next/adapters/containerimage/pull.go

```diff
@@ -93,7 +93,11 @@ func (is *Source) resolveRemote(ctx context.Context, ref string, platform *ocisp
 		dgst digest.Digest
 		dt   []byte
 	}
-	res, err := is.g.Do(ctx, ref, func(ctx context.Context) (interface{}, error) {
+	p := platforms.DefaultSpec()
+	if platform != nil {
+		p = *platform
+	}
+	res, err := is.g.Do(ctx, "getconfig::"+ref+"::"+platforms.Format(p), func(ctx context.Context) (interface{}, error) {
```

Comment:
I have no idea what this is doing, likewise in `resolve`. What is this format? What affect does it have?

Reply:
> Do executes a context function syncronized by the key

https://godoc.org/github.com/moby/buildkit/util/flightcontrol#Group.Do

If the key string is not per-platform, the callback running for a specific platform prevents the run for other platforms when requested simultaneously, that's what I understand at least.

Reply:
Can we make a comment for this, or a function like `func platformKey(op, ref string, platform platform.Platform)`